common/transport/http/src/rest_api_client.ts

```diff
@@ -37,6 +37,7 @@ export class RestApiClient {
   private _config: RestApiClient.TransportConfig;
   private _http: HttpBase;
   private _userAgent: string;
+  private _currentRequest: ClientRequest;
```

Comment:
>_currentRequest [](start = 10, length = 15)

this assumes there's only one request at a time - this assumption is not clear at all from the outside perspective (looking at the RestApiClient API).